File: SubscriptionApi/SubscriptionApi/Commands/SubscriptionStopCommandFromApi.cs
using SubscriptionApi.Dto;
using SubscriptionApi.Exceptions;
using SubscriptionApi.ResponseBuilders;
using SubscriptionApi.Subscriptions;

namespace SubscriptionApi.Commands
{
  public interface SubscriptionStopCommand
  {
    StopSubscriptionResponseDto Response(); //bug is this used?
    void ValidateData();
    void Authorize();
    void Invoke();
  }

  public class SubscriptionStopCommandFromApi : SubscriptionStopCommand
  {
    private readonly StoppedSubscriptionParametersDto _parameters;
    private readonly SubscriptionStopResponseBuilder _responseBuilder;
    private readonly SubscriptionsModifyOperations _subscriptions;
    private readonly SubscriptionDataCorrectnessCriteria _correctnessCriteria;
    private readonly UserAuthorization _userAuthorization;

    public SubscriptionStopCommandFromApi(
      StoppedSubscriptionParametersDto parameters, 
      SubscriptionStopResponseBuilder responseBuilder, 
      SubscriptionsModifyOperations subscriptions, 
      SubscriptionDataCorrectnessCriteria correctnessCriteria, 
      UserAuthorization userAuthorization)
    {
      _parameters = parameters;
      _responseBuilder = responseBuilder;
      _subscriptions = subscriptions;
      _correctnessCriteria = correctnessCriteria;
      _userAuthorization = userAuthorization;
    }

    public StopSubscriptionResponseDto Response()
    {
      return _responseBuilder.Build();
    }

    public void ValidateData()
    {
      _correctnessCriteria.ValidateSessionId(_parameters.SubscriptionId, _responseBuilder);
      _correctnessCriteria.ValidateUserName(_parameters.UserName, _responseBuilder);
      _responseBuilder.AssertNoFatalErrors(new FatalErrorOccuredDuringValidationsException());
    }

    public void Authorize()
    {
      _userAuthorization.VerifyUserExistence(_parameters.UserName, _responseBuilder);
    }

    public void Invoke()
    {
      _subscriptions.Remove(_parameters.SubscriptionId, _responseBuilder);
    }
  }
}

File: Java/factories/_01_no_factories/src/main/java/Commands/SubscriptionStopCommandFromApi.java
package commands;

import dto.StopSubscriptionResponseDto;
import dto.StoppedSubscriptionParametersDto;
import exceptions.FatalErrorOccuredDuringValidationsException;
import responseBuilders.SubscriptionStopResponseBuilder;
import subscriptions.DataCorrectnessCriteria;
import subscriptions.SubscriptionsModifyOperations;

public class SubscriptionStopCommandFromApi
    implements SubscriptionCommand {
    private final StoppedSubscriptionParametersDto parameters;
    private final SubscriptionStopResponseBuilder responseBuilder;
    private final SubscriptionsModifyOperations subscriptions;
    private final DataCorrectnessCriteria correctnessCriteria;
    private final UserAuthorization userAuthorization;

    public SubscriptionStopCommandFromApi(
        final StoppedSubscriptionParametersDto parameters,
        final SubscriptionStopResponseBuilder responseBuilder,
        final SubscriptionsModifyOperations subscriptions,
        final DataCorrectnessCriteria correctnessCriteria,
        final UserAuthorization userAuthorization) {

        this.parameters = parameters;
        this.responseBuilder = responseBuilder;
        this.subscriptions = subscriptions;
        this.correctnessCriteria = correctnessCriteria;
        this.userAuthorization = userAuthorization;
    }

    public StopSubscriptionResponseDto response() {
        return responseBuilder.buildStop();
    }

    public void validateData() {
        correctnessCriteria.validateSessionId(parameters.subscriptionId, responseBuilder);
        correctnessCriteria.validateUserName(parameters.userName, responseBuilder);
        responseBuilder.assertNoFatalErrors(new FatalErrorOccuredDuringValidationsException());
    }

    public void authorize() {
        userAuthorization.verifyUserExistence(parameters.userName, responseBuilder);
    }

    public void invoke() {
        subscriptions.remove(parameters.subscriptionId, responseBuilder);
    }

    public void resolve() {

    }
}
